openedx/core/lib/xblock_serializer/test_api.py

```diff
@@ -287,3 +415,273 @@ def test_jsinput_extra_files(self):
+            f"""
+            <problem
+                display_name="Tagged Python Problem"
+                url_name="Tagged_Python_Problem"
+                tags-v1="{expected_serialized_tags}"
+            >
+                This uses python: <script type="text/python">...</script>...
+            </problem>
+            """
+        )
+
+    def test_tagged_library_content_blocks(self):
+        """
+        Test library content blocks that have applied tags
+        """
+        course = CourseFactory.create(display_name='Tagged Library Content course', run="TLCC")
+        lib = LibraryFactory()
+        lc_block = BlockFactory(
+            parent_location=course.location,
+            category="library_content",
+            source_library_id=str(lib.location.library_key),
+            display_name="Tagged LC Block",
+            max_count=1,
+        )
+
+        # Add a bunch of tags to the library content block
+        tagging_api.tag_object(
+            object_id=lc_block.location,
+            taxonomy=self.taxonomy1,
+            tags=["normal tag", "<special \"'-=,. |= chars > tag", "anotherTag"]
+        )
+
+        # Check that the tags data in included in the OLX and properly escaped
+        serialized = api.serialize_xblock_to_olx(lc_block)
+        self.assertXmlEqual(
+            serialized.olx_str,
+            f"""
+            <library_content
+                display_name="Tagged LC Block"
+                max_count="1"
+                source_library_id="{str(lib.location.library_key)}"
+                url_name="Tagged_LC_Block"
+                tags-v1="t1-export-id:%3Cspecial %22%27-%3D%2C. %7C%3D chars %3E tag,anotherTag,normal tag"
+            />
+            """
+        )
+
+    def test_tagged_video_block(self):
+        """
+        Test video blocks that have applied tags
+        """
+        course = CourseFactory.create(display_name='Tagged Video Test course', run="TVTC")
+        video_block = BlockFactory.create(
+            parent_location=course.location,
+            category="video",
+            display_name="Tagged Video Block",
+        )
+
+        # Add tags to video block
+        tagging_api.tag_object(
+            object_id=video_block.location,
+            taxonomy=self.taxonomy1,
+            tags=["normal tag", "<special \"'-=,. |= chars > tag", "anotherTag"]
+        )
+
+        # Check that the tags data in included in the OLX and properly escaped
+        serialized = api.serialize_xblock_to_olx(video_block)
+        self.assertXmlEqual(
+            serialized.olx_str,
+            """
+            <video
+                youtube="1.00:3_yD_cEKoCk"
+                url_name="Tagged_Video_Block"
+                display_name="Tagged Video Block"
+                tags-v1="t1-export-id:%3Cspecial %22%27-%3D%2C. %7C%3D chars %3E tag,anotherTag,normal tag"
+            />
+            """
+        )
+
+    def test_tagged_openassessment_block(self):
```

Comment:
@yusuf-musleh @bradenmacdonald @ChrisChV 

This test, relies on the openassessment block which comes from ora2 and having this test here caused an interesting issue.  When new attributes got added to the block in the edx-ora2 repo, it broke this test because the expected OLX in this repo did not have the new attributes.  This seems like bad coupling across repos and my instinct would be to move/remove this specific test.  But I'm not sure what the specific goals of this test and how important it is.  Can we remove this from the edx-platform repo?

Reply:
@feanil The implementation/tests got updated since and now tags are no longer serialized and included in the actual OLX. So we could actually change the test case, namely changing [this assert](https://github.com/openedx/edx-platform/blob/master/openedx/core/lib/xblock_serializer/test_api.py#L704-L707) to just check that there is no tags data in the OLX, and we can get rid of `EXPECTED_OPENASSESSMENT_OLX` since we would no longer need it.

That way it wouldn't cause issues when attributes are added/removed from the ora2, and avoids removing the test case, does that sound like a reasonable approach?

Reply:
@yusuf-musleh that sounds great, please make that change and add a link to the PR here.

Reply:
@feanil Sure thing, here it is https://github.com/openedx/edx-platform/pull/34782